osu.Game/Screens/Edit/Editor.cs

```diff
@@ -541,10 +547,24 @@ protected void Cut()
 
         protected void Copy()
         {
+            var builder = new StringBuilder();
```

Comment:
I don't think all of this clipboard string constructing logic needs to be at the very root editor level. Ideally it should be in a component that's 99% isolated from the editor (aside from the one necessary `Add()` call). All dependencies that this logic needs should be reachable via DI.

If that can't be done due to keybinding handling foibles, then I'd take a single parameterlesw function call on top of the add, too.